site/src/docs/patterns/cookies/basics.mdx

```diff
@@ -0,0 +1,114 @@
+| **Social media**              | Social media cookies are related to the characteristics of various social media services.                                      | Never necessary.                                                                                                                                                                                                                                                                             | Social media feeds, social media share buttons                                               |
+| **Accessibility**             | Accessibility cookies enable some of the service’s accessibility features.                                                     | Always necessary.                                                                                                                                                                                                                                                                            | Accessibility related preferences such as audio descriptions                                 |
+| **Statistics**                | The information collected by statistics cookies is used to develop the website.                                                | Usually not necessary. If analytics are considered absolutely necessary for the service, cookies can be considered essential and this must be informed to the user. When considered essential, this must be justified to the user. Gathered information must not be handed to third parties. | Unique visitor IP addresses, page views count, measurements of how the user uses the service |
+| **Device information**        | Device information cookies contain information collected from the user’s device that ensures the functionality of the service. | Never necessary.                                                                                                                                                                                                                                                                             | Device information, OS, and browser information                                              |
+| **Chat services**             | Chat service cookies enable communication between the user and the service.                                                    | Can only be considered necessary if real-time communication is the main purpose of the service. When the user opens the chat window, cookies related to it become necessary. Note that they are not necessary until the user opens the chat.                                                 | Chat provider cookies                                                                        |
```

Comment:
These two sentences seems to be contradicting each other:

> Can only be considered necessary if real-time communication is the main purpose of the service. When the user opens the chat window, cookies related to it become necessary.

If a service has a chat window, it rarely is the main purpose of the service. So according to the first sentence, the chat cookies should then not be considered necessary. But according to the second sentence, they still have the option to become necessary. And we only ask cookies when the page loads. So I don't think we can change cookie from optional to necessary during the web page visit.

Reply:
Optional cookies **could be** changed to required or new cookies **could be** added, but that would require re-rendering the cookie component with new props. This is quite hard to implement currently, but not impossible. Probably should not be documented. :)